### PR TITLE
Fix mv rewrite bug

### PR DESCRIPTION
**Pull request by tinkerrrr**
## Proposed changes

Close related [#7361](https://github.com/apache/incubator-doris/issues/7361)

As the sql described in [#7361](https://github.com/apache/incubator-doris/issues/7361)
```
select k1, count(k2) / count(1) from UserTable group by k1
``` 

Before this pr, `count(k2) / count(1)` will be rewritten as `sum(UserTable.mv_count_k2) / count(1)`, and will be kept in second-round analyze, which could cause mv select fail.

After this pr, `count(k2) / count(1)` will still be rewritten as `sum(UserTable.mv_count_k2) / count(1)`, but won't be kept in second-round analyze, so query could successfully run.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by EmmyMiao87** [comment]
The unit test is failed 'testSubqueryInCase'

**Comment by tinkerrrr** [comment]
@EmmyMiao87 UT fixed, PTAL

File: fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java [hunk 1]
```diff
@@ -673,6 +673,17 @@ private void resetAnalyzerAndStmt() {
         analyzer = new Analyzer(context.getCatalog(), context);
 
         parsedStmt.reset();
+
+        if (parsedStmt instanceof SelectStmt) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java:677]: Could you add comment to explain why we need to reset select list here?
And SelectStmt is one kind of QueryStmt, what about SetOperationStmt?

Comment (morningman) [reply fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java:677]: And release rebase the master to pass the UT

Comment (tinkerrrr) [reply fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java]: Done, SetOperationStmt need to handle too

**Comment by tinkerrrr** [comment]
`MaterializedViewFunctionTest.testQueryOnStarAndJoin` failed but could pass on local. Need some help @morningman

**Comment by morningman** [comment]
hi @tinkerrrr , sorry for the delay.
You can just delete the `testQueryOnStarAndJoin()` test, because it is meaningless.
Please help us to delete it, and than rebase the master again. Thank you

**Comment by tinkerrrr** [comment]
> hi @tinkerrrr , sorry for the delay. You can just delete the `testQueryOnStarAndJoin()` test, because it is meaningless. Please help us to delete it, and than rebase the master again. Thank you

@morningman Done

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**